cs4660/graph/graph.py

```diff
@@ -0,0 +1,289 @@
+* remove_edge
+    - remove an edge from its internal data structure
+    - returns true if the edge is removed and false if the edge does not exist
+"""
+
+from io import open
+from operator import itemgetter
+
+def construct_graph_from_file(graph, file_path):
+    with open(file_path) as file:
+        next(file)
+        for line in file:
+            graph.add_edge(getEdge(line))
+    return graph
+
+def getEdge(line):
+    value = line.strip().split(":")
+    return Edge(Node(int(value[0])), Node(int(value[1])), int(value[2]))
+
+class Node(object):
+    """Node represents basic unit of graph"""
+    def __init__(self, data):
+        self.data = data
+
+    def __str__(self):
+        return 'Node({})'.format(self.data)
+
+    def __repr__(self):
+        return 'Node({})'.format(self.data)
+
+    def __eq__(self, other_node):
+        return self.data == other_node.data
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __hash__(self):
+        return hash(self.data)
+
+class Edge(object):
+    """Edge represents basic unit of graph connecting between two edges"""
+    def __init__(self, from_node, to_node, weight):
+        self.from_node = from_node
+        self.to_node = to_node
+        self.weight = weight
+
+    def __str__(self):
+        return 'Edge(from {}, to {}, weight {})'.format(self.from_node, self.to_node, self.weight)
+
+    def __repr__(self):
+        return 'Edge(from {}, to {}, weight {})'.format(self.from_node, self.to_node, self.weight)
+
+    def __eq__(self, other_node):
+        return self.from_node == other_node.from_node and self.to_node == other_node.to_node and self.weight == other_node.weight
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __hash__(self):
+        return hash((self.from_node, self.to_node, self.weight))
+
+
+class AdjacencyList(object):
+    """
+    AdjacencyList is one of the graph representation which uses adjacency list to
+    store nodes and edges
+    """
+    def __init__(self):
+        # adjacencyList should be a dictonary of node to edges
+        self.adjacency_list = {}
+
+    def adjacent(self, node_1, node_2):
+        if node_1 in self.adjacency_list:
+            allEdges = []
+            for edge in self.adjacency_list[node_1]:
+                allEdges.append(edge.to_node)
+            return node_2 in allEdges
+        return False
+
+    def neighbors(self, node):
+        neighbors = []
+        if node in self.adjacency_list:
+            for edge in self.adjacency_list[node]:
+                neighbors.append(edge.to_node)
+        return neighbors
+
+    def add_node(self, node):
+        if node not in self.adjacency_list:
+            self.adjacency_list[node] = []
+            return True
+        return False
+
+    def remove_node(self, node):
+        for nodes in self.adjacency_list:
+            for edge in self.adjacency_list[nodes]:
+                if edge.to_node.__eq__(node):
```

Comment:
not sure why you need to call `__eq__` method here. You can simply use `==` and implicitly it will call `__eq__` method.